### ISSUE TITLE
Parameters to proxy pac URL

### ISSUE THREAD

**Issue by jeka** [FelisCatus/SwitchyOmega#1000]
Hi,

I am not sure what is the cause and this  was working until recently but for some reason it doesn't work anymore. Yes, I see that you haven't done release for a while but  maybe something with Chrome got changed recently?
 
The problem: My company uses on fly proxy.pac generator and the result depends on IP from which you download the file. Also, you can specify IP as parameter to the URL and proxy.pac file is generated for the specified IP.
Until recently I have used URL like: http://proxyconf/proxy.pac\?10.159.12.1 (notice the escaped ?) and it was working perfectly. Now I am getting HTTP/403 error when I hit "Download profile now". 
If I don't escape '?' I don't get error but parameter is not passed correctly it seems as I don't get the desired effect. The returned PAC file is as for my current IP not the one specified.

I have no idea when exactly this got broken. I just noticed it today ...

Just for reference both command with curl in terminal from the same machine:
**1. with parameter**
```
curl -s --noproxy '\*' http://proxyconf/proxy.pac?10.159.12.1  | head -5
/*
** automatic proxy configuration V2.0 for 10.159.12.1 from proxyconf (10.158.55.18)
*/
function FindProxyForURL(url, host) {
 var proxy="PROXY fi-proxy.ddd.net:8080; PROXY de-proxy.ddd.net:8080";
```
**2. without parameter**
```
curl -s --noproxy '\*' http://proxyconf/proxy.pac  | head -5
/*
** automatic proxy configuration V2.0 for 10.154.162.60 from proxyconf (10.159.43.54)
*/
function FindProxyForURL(url, host) {
 var proxy="PROXY hu-proxy.ddd.net:8080; PROXY fi-proxy.ddd.net:8080";
```
So question: Can I see somehow what exactly request **SwitchyOmega** is making so I get this 403 error?

**Comment by FelisCatus** [comment]
To work around caches, SwitchyOmega trys to add a dummy parameter such as `?_=1488258904500` to your PAC URL when you click on the button. (Before that, people kept complaining that button does not clear the cache.) Automatic updates do not have that parameter added so I don't think there is a problem with that.

To anticipate the problem of some servers rejecting the request when an extra parameter is passed, SwitchyOmega will retry the request without the parameter if it gets an error response. If that also gives your a 403 error, then it gives up.

For now I have no idea on what is the problem. But it is certainly possible to check the requests. You just need to go to `chrome://extensions/?id=padekgcemlokbadohgkifijomclgjgif`, enable Developer mode and  click on the `background.html` under SwitchyOmega. Now in the opened developer tools, choose the `Network` tab. From now on, every request would be shown on this window, so don't close it. Now if you click on the `Download profile now` button on the options page, a record would be there.

If you see something suspicious, for example the first one fails but there is no second request, please let me know so that I can take a look.

**Comment by jeka** [comment]
Thanks for the reply and debug directions.
It seem SwitchyOmega is little bit buggy with regard of adding this extra parameter:
If I put ```http://proxyconf.emea.nsn-net.net/proxy.pac?10.159.12.1``` as PAC URL
the resulting request is
```http://proxyconf.emea.nsn-net.net/proxy.pac?10.159.12.1=&_=1488271584257```
(too many equal signs there).

Anyway, it would not work in my case anyway. What I need is SwitchyOmega to use the exact URL I have provided. No additions in any form or shape.

Is it possible this adding of dummy parameter to be made configurable? Or even better: analyze URL and if there are any parameters already don't alter it at all.

**Comment by FelisCatus** [comment]
I will go with the analysis path then.

**Comment by FelisCatus** [comment]
Now it won't touch the URL when there's a `?` in it to be on the safe side. I'll probably do a release later. Stay tuned.

**Comment by FelisCatus** [comment]
Fixed in v2.4.2

**Comment by jeka** [comment]
No problem for the delay. Now its working as expected in my case.
Thank you!
